dev/system-test/firestore.ts

```diff
@@ -3844,6 +3625,236 @@ describe.skip('Aggregation queries', () => {
+        for (const docPath of docPaths) {
+          batch.set(firestore.doc(docPath), {x: 2});
+        }
+        await batch.commit();
+        const snapshot = await firestore
+          .collectionGroup(collectionGroupId)
+          .aggregate({
+            count: AggregateField.count(),
+            sum: AggregateField.sum('x'),
+            avg: AggregateField.average('x'),
+          })
+          .get();
+        expect(snapshot.data().count).to.equal(2);
+        expect(snapshot.data().sum).to.equal(4);
+        expect(snapshot.data().avg).to.equal(2);
+      });
+
+      it('performs aggregations on documents with all aggregated fields', async () => {
+        const testDocs = {
+          a: {author: 'authorA', title: 'titleA', pages: 100, year: 1980},
+          b: {author: 'authorB', title: 'titleB', pages: 50, year: 2020},
+          c: {author: 'authorC', title: 'titleC', pages: 150, year: 2021},
+          d: {author: 'authorD', title: 'titleD', pages: 50},
+        };
+        await addTestDocs(testDocs);
+        const snapshot = await col
+          .aggregate({
+            totalPages: AggregateField.sum('pages'),
+            averagePages: AggregateField.average('pages'),
+            averageYear: AggregateField.average('year'),
+            count: AggregateField.count(),
+          })
+          .get();
+        expect(snapshot.data().totalPages).to.equal(300);
+        expect(snapshot.data().averagePages).to.equal(100);
+        expect(snapshot.data().averageYear).to.equal(2007);
+        expect(snapshot.data().count).to.equal(3);
+      });
+
+      it('performs aggregates on multiple fields where one aggregate could cause short-circuit due to NaN', async () => {
+        const testDocs = {
+          a: {
+            author: 'authorA',
+            title: 'titleA',
+            pages: 100,
+            year: 1980,
+            rating: 5,
+          },
+          b: {
+            author: 'authorB',
+            title: 'titleB',
+            pages: 50,
+            year: 2020,
+            rating: 4,
+          },
+          c: {
+            author: 'authorC',
+            title: 'titleC',
+            pages: 100,
+            year: 1980,
+            rating: Number.NaN,
+          },
+          d: {
+            author: 'authorD',
+            title: 'titleD',
+            pages: 50,
+            year: 2020,
+            rating: 0,
+          },
+        };
+        await addTestDocs(testDocs);
+        const snapshot = await col
+          .aggregate({
+            totalRating: AggregateField.sum('rating'),
+            totalPages: AggregateField.sum('pages'),
+            averageYear: AggregateField.average('year'),
+          })
+          .get();
+        expect(snapshot.data().totalRating).to.be.NaN;
+        expect(snapshot.data().totalPages).to.equal(300);
+        expect(snapshot.data().averageYear).to.equal(2000);
+      });
+
+      it('performs aggregates when using `in` operator', async () => {
```

Comment:
This test can be modified to run against production by dropping the aggregations over fields other than `'rating'`. For your consideration.

Reply:
Done